src/numbers/RequestErrorDetector.java

```diff
@@ -0,0 +1,91 @@
+            final String[] incorrectProperties = getIncorrectProperties(numberProperties);
+            final boolean hasOneWrongProperty = incorrectProperties.length == 1;
+            final int currentErrorMessageIndex = hasOneWrongProperty
+                    ? errorMessageIndex.PROPERTY : errorMessageIndex.PROPERTIES;
+
+            errorMessage= String.format(ERROR_MESSAGES[currentErrorMessageIndex], String.join(" ", incorrectProperties));
+        } else {
+            final var propertiesList = List.of(numberProperties.split(" "));
+            final String requestedMutuallyExclusiveProperties = getMutuallyExclusiveProperties(propertiesList);
+
+            errorMessage = requestedMutuallyExclusiveProperties.isEmpty()
+                    ? "" : String.format(ERROR_MESSAGES[errorMessageIndex.MUTUALLY_EXCLUSIVE_PROPERTIES],
+                    requestedMutuallyExclusiveProperties);
+        }
+    }
+
+    private String[] getIncorrectProperties(final String numberProperties) {
+        return Stream.of(numberProperties.split(" "))
+                .filter(np -> Stream.of(Properties.AVAILABLE_PROPERTIES).noneMatch(ap -> np.matches("-?" + ap)))
+                .toArray(String[]::new);
+    }
+
+    private String getMutuallyExclusiveProperties(List<String> numberProperties) {
+        final var assertedProperties= numberProperties.stream().filter(np -> !np.startsWith("-")).toList();
+        final var negatedProperties = numberProperties.stream().filter(np -> np.startsWith("-")).toList();
+
+        final String polarisedProperties = negatedProperties.stream()
+                .filter(np -> String.join("", assertedProperties).contains(np.replaceFirst("-", "")))
+                .map(np -> np.replaceFirst("-", "") + " & " + np)
+                .collect(Collectors.joining(", "));
+        final String assertedMutuallyExclusiveProperties = Stream.of(Properties.MUTUALLY_EXCLUSIVE_PROPERTIES)
+                .filter(mep -> assertedProperties.contains(mep[0]) && assertedProperties.contains(mep[1]))
+                .map(mep -> mep[0] + " & " + mep[1])
+                .collect(Collectors.joining(", "));
+        final String negatedMutuallyExclusiveProperties = Stream.of(Properties.MUTUALLY_EXCLUSIVE_PROPERTIES)
+                .filter(mep -> negatedProperties.contains('-' + mep[0]) && negatedProperties.contains('-' + mep[1]))
+                .map(mep -> '-' + mep[0] + " & -"  + mep[1])
+                .collect(Collectors.joining(", "));
+
+        return (polarisedProperties + ", "
+                + assertedMutuallyExclusiveProperties + ", "
+                + negatedMutuallyExclusiveProperties)
+                .replaceFirst("(?:, )*$", "").replaceFirst("^(?:, )*", "");
+    }
+
+    public String getErrorMessage() {
+        return errorMessage;
+    }
+
+    public boolean isValidRequest() {
+        return errorMessage.isEmpty();
+    }
+
+    private static class errorMessageIndex {
```

Comment:
Weird class name. And why do you need a class for constants?

Reply:
It may sound weird, but I declared class for constatnts to make code more readable.